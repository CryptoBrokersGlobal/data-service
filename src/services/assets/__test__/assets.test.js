const { get, mget, search } = require('../sql');

describe('Assets SQL queries tests', () => {
  it('should build correct get sql query', () => {
    expect(get('123')).toMatchSnapshot();
  });

  it('should build correct mget sql query', () => {
    expect(mget(['1', '2', '3'])).toMatchSnapshot();
  });

  it('should build correct search sql query', () => {
    expect(search({ ticker: '*' })).toMatchSnapshot();
  });
});
