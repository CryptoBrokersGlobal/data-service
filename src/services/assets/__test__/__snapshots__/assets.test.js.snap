// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets SQL queries tests should build correct get sql query 1`] = `"select \\"asset_id\\", \\"asset_name\\", \\"description\\", \\"sender\\", \\"issue_height\\", \\"total_quantity\\", \\"decimals\\", \\"reissuable\\", \\"ticker\\", \\"issue_timestamp\\", \\"has_script\\" from \\"assets\\" where \\"asset_id\\" in ('123')"`;

exports[`Assets SQL queries tests should build correct mget sql query 1`] = `"select \\"asset_id\\", \\"asset_name\\", \\"description\\", \\"sender\\", \\"issue_height\\", \\"total_quantity\\", \\"decimals\\", \\"reissuable\\", \\"ticker\\", \\"issue_timestamp\\", \\"has_script\\" from \\"assets\\" where \\"asset_id\\" in ('1', '2', '3')"`;

exports[`Assets SQL queries tests should build correct search sql query 1`] = `"select \\"asset_id\\", \\"asset_name\\", \\"description\\", \\"sender\\", \\"issue_height\\", \\"total_quantity\\", \\"decimals\\", \\"reissuable\\", \\"ticker\\", \\"issue_timestamp\\", \\"has_script\\" from \\"assets\\" where \\"ticker\\" is not null"`;
